flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/JoinTestPrograms.java

```diff
@@ -0,0 +1,450 @@
+                    .producedAfterRestore(
+                            Row.of(4, 3L, "Juliet"),
+                            Row.of(4, 4L, "Helena"),
+                            Row.of(1, 1L, "Ivana"))
+                    .build();
+
+    static final SourceTestStep DEPARTMENT =
+            SourceTestStep.newBuilder("DEPARTMENT")
+                    .addSchema(
+                            "department_num int", "b2 bigint", "b3 int", "department_name varchar")
+                    .producedBeforeRestore(
+                            Row.of(null, 102L, 0, "Accounting"),
+                            Row.of(1, 1L, 0, "Research"),
+                            Row.of(2, 2L, 1, "Human Resources"),
+                            Row.of(2, 3L, 2, "HR"),
+                            Row.of(3, 1L, 2, "Sales"))
+                    .producedAfterRestore(
+                            Row.of(2, 4L, 3, "People Operations"), Row.of(4, 2L, 4, "Engineering"))
+                    .build();
+
+    static final SourceTestStep DEPARTMENT_NONULLS =
+            SourceTestStep.newBuilder("DEPARTMENT")
+                    .addSchema(
+                            "department_num int", "b2 bigint", "b3 int", "department_name varchar")
+                    .producedBeforeRestore(
+                            Row.of(1, 1L, 0, "Research"),
+                            Row.of(2, 2L, 1, "Human Resources"),
+                            Row.of(2, 3L, 2, "HR"),
+                            Row.of(3, 1L, 2, "Sales"))
+                    .producedAfterRestore(Row.of(2, 4L, 3, "People Operations"))
+                    .build();
+    static final SourceTestStep SOURCE_T1 =
+            SourceTestStep.newBuilder("T1")
+                    .addSchema("a int", "b bigint", "c varchar")
+                    .producedBeforeRestore(
+                            Row.of(1, 1L, "Baker1"),
+                            Row.of(1, 2L, "Baker2"),
+                            Row.of(1, 2L, "Baker2"),
+                            Row.of(1, 5L, "Baker3"),
+                            Row.of(2, 7L, "Baker5"),
+                            Row.of(1, 9L, "Baker6"),
+                            Row.of(1, 8L, "Baker8"),
+                            Row.of(3, 8L, "Baker9"))
+                    .producedAfterRestore(Row.of(1, 1L, "PostRestore"))
+                    .build();
+    static final SourceTestStep SOURCE_T2 =
+            SourceTestStep.newBuilder("T2")
+                    .addSchema("a int", "b bigint", "c varchar")
+                    .producedBeforeRestore(
+                            Row.of(1, 1L, "BakerBaker"),
+                            Row.of(2, 2L, "HeHe"),
+                            Row.of(3, 2L, "HeHe"))
+                    .producedAfterRestore(Row.of(2, 1L, "PostRestoreRight"))
+                    .build();
+
+    static {
+        NON_WINDOW_INNER_JOIN =
+                TableTestProgram.of("non-window-inner-join", "test non-window inner join")
```

Comment:
I missed it previously. Could you please adapt the id according to:
https://github.com/apache/flink/blob/52d8d3583e5c989da84126a8805ab335408c46c2/flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/test/program/TableTestProgram.java#L117

Reply:
Yes!